beamer/agent/middleware.py

```diff
@@ -39,13 +68,13 @@ def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
             response = make_request(method, params)
             if _result_ok(response):
                 key = hex(response["result"].number), params[1]
-                cache[key] = response
+                _BLOCK_STORAGE[chain_id].add_block(key, response)
         elif params[0].startswith("0x"):
-            response = cache.get(params)
+            response = _BLOCK_STORAGE[chain_id].get_block(params)
```

Comment:
do you know if we have to set the alternative (param `instead`)?

Reply:
imo params is key in the latest part

Reply:
sorry, I meant the function `lru.get(key, instead)`. I know for dicts its returning None per default at key error. Is that also the case for lru ?

Reply:
yes.